src/Facility.Core/ServiceSerializer.cs

```diff
@@ -0,0 +1,66 @@
+using System.Diagnostics.CodeAnalysis;
+
+namespace Facility.Core;
+
+/// <summary>
+/// Serializes and deserializes values to and from a serialization format.
+/// </summary>
+public abstract class ServiceSerializer
+{
+	/// <summary>
+	/// Serializes a value to the serialization format.
+	/// </summary>
+	public abstract string ToString(object? value);
+
+	/// <summary>
+	/// Serializes a value to the serialization format.
+	/// </summary>
+	public abstract void ToStream(object? value, Stream outputStream);
+
+	/// <summary>
+	/// Serializes a value to the serialization format.
+	/// </summary>
+	public virtual Task ToStreamAsync(object? value, Stream outputStream, CancellationToken cancellationToken)
+	{
+		ToStream(value, outputStream);
+		return Task.CompletedTask;
+	}
+
+	/// <summary>
+	/// Deserializes a value from the serialization format.
+	/// </summary>
+	public abstract object? FromString(string stringValue, Type type);
+
+	/// <summary>
+	/// Deserializes a value from the serialization format.
+	/// </summary>
+	public T? FromString<T>(string stringValue) => (T?) FromString(stringValue, typeof(T));
```

Comment:
`virtual`, perhaps, in case the generic form is optimized?

Reply:
Ha, I had it `virtual` at first, then removed it.